packages/demos/stories/DragDrop.tsx

```diff
@@ -0,0 +1,200 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayList from '@clayui/list';
+import {XYCoord} from 'dnd-core';
+import React, {useState, useCallback, useRef} from 'react';
+import {useDrag, useDrop, DropTargetMonitor} from 'react-dnd';
+
+interface IDraggableListItemProps {
+	/**
+	 * List Item unique identifier
+	 */
+	id: any;
+
+	/**
+	 * The text displayed in the List Item
+	 */
+	text: string;
+
+	/**
+	 * Index of the List Item in order to keep track of it's positioning within the array
+	 */
+	index: number;
+
+	/**
+	 * Function that handles the moving of List Items
+	 */
+	onMove: (dragIndex: number, hoverIndex: number) => void;
+}
+
+interface IDragItem {
+	/**
+	 * List Item unique identifier
+	 */
+	id: string;
+
+	/**
+	 * Index of the List Item in order to keep track of it's positioning within the array
+	 */
+	index: number;
+
+	/**
+	 * Type of data that is allowed to be dragged
+	 */
+	type: string;
+}
+
+const DraggableListItem: React.FunctionComponent<IDraggableListItemProps> = ({
+	id,
+	index,
+	onMove,
+	text,
+}) => {
+	const ref = useRef<HTMLDivElement>(null);
+
+	const [, drop] = useDrop({
+		accept: 'listItem',
+		hover(item: IDragItem, monitor: DropTargetMonitor) {
+			const dragIndex = item.index;
+
+			const hoverIndex = index;
+
+			if (!ref.current || dragIndex === hoverIndex) {
+				return;
+			}
+
+			const hoverBoundingRect = ref.current!.getBoundingClientRect();
+
+			const verticalMiddle =
+				(hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+			const mousePosition = monitor.getClientOffset();
+
+			const pixelsToTop =
+				(mousePosition as XYCoord).y - hoverBoundingRect.top;
+
+			const draggingDownwards =
+				dragIndex < hoverIndex && pixelsToTop < verticalMiddle / 2;
+
+			if (draggingDownwards) {
```

Comment:
Lets move this below `draggingUpwards ` and that way we can condense the logic to be

```js
if (draggingDownwards || draggingUpwards) {
    return;
} 
```

I think this might help readability a bit